src/v/raft/follower_stats.h

```diff
@@ -76,6 +77,17 @@ class follower_stats {
 
     void update_with_configuration(const group_configuration&);
 
+    void reset() {
+        for (auto& [_, meta] : _followers) {
+            meta.last_dirty_log_index = model::offset{};
+            meta.last_flushed_log_index = model::offset{};
+            meta.last_sent_offset = model::offset{};
+            meta.match_index = model::offset{};
+            meta.next_index = model::offset{};
+            meta.heartbeats_failed = 0;
+        }
```

Comment:
kinda seems like you should implementation `follower_index_metadata::reset` (or whatever method name makes sense) to keep methods on that data with the structure itself?